src/util/bv_encoding.cpp

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************\
+
+Module: bv_encoding configuration
+
+Author: Diffblue Ltd
+
+\*******************************************************************/
+
+#include "bv_encoding.h"
+
+#include "exception_utils.h"
+#include "string2int.h"
+
+bv_encodingt parse_object_bits_encoding(
+  const std::string &argument,
+  const std::size_t pointer_width)
+{
+  const unsigned int object_bits = unsafe_string2unsigned(argument);
```

Comment:
Use `string2optional` here so you can handle a parse error with a proper error message. The `unsafe_` functions are for when you know the parse will succeed.

Reply:
Yep. A non-integer argument will cause an invariant violation, both before and after this PR. This is expected as the PR is intended to be refactoring only, not bug fixing. I agree that we should improve the error handling here.